### PR TITLE
[next] Initial next branch cleanup

### PR DESCRIPTION
**Pull request by michealparks**
Just gettin' some more tedious stuff out of the way 🧹 

This PR mostly consists of repetitive cleanup following the initial next branch:
* Migrate more components to runes mode. All that's left after this is `theater`, `xr`, and some `extras` that need care.
* Fix incorrect type issues with `ref` bindings with many extras components.
* Remove usage of `$$Props`, since the `$props` rune makes it unnecessary.
* Make sure all repos are using the latest and same dependency versions.

And a few one-off fixes:
* Fix animated sprite material bug.
* Add a few missing component `ref` bindings.

### CODE REVIEW

File: packages/extras/src/lib/components/ContactShadows/ContactShadows.svelte [hunk 1]
```diff
@@ -114,43 +102,46 @@
...
-    transparent: true,
-    opacity,
-    depthWrite
-  })
 
   const blurShadows = (blur: number) => {
     // separate from store to not call store setter
-    const bp = $blurPlane
-    bp.visible = true
+    blurPlane.visible = true
```
Comment (DefinitelyMaybe) [inline packages/extras/src/lib/components/ContactShadows/ContactShadows.svelte:128]: `bp` -> `blurplane`, thank you! 🙏

File: packages/extras/src/lib/components/InstancedSprite/InstancedSprite.svelte [hunk 2]
```diff
@@ -122,39 +113,37 @@
...
 <T
-  is={ref}
+  is={mesh}
+  bind:ref
   frustumCulled={false}
-  {...$$restProps}
+  {...props}
 >
-  <slot Instance={proxySpritefileComponent} />
+  <slot Instance={SpriteInstance} />
```
Comment (DefinitelyMaybe) [inline packages/extras/src/lib/components/InstancedSprite/InstancedSprite.svelte:148]: 👍

File: packages/extras/src/lib/components/Instancing/InstancedMeshes/InstancedMeshes.svelte [hunk 3]
```diff
@@ -43,17 +42,19 @@
...
+  let meshesArray = $derived(Array.isArray(meshes) ? meshes : Object.values(meshes))
+  let filteredMeshesArray = $derived(meshesArray.filter((mesh) => mesh.isMesh))
 </script>
 
 <InnerInstancedMeshes
   meshes={filteredMeshesArray}
-  {...$$restProps}
+  {...props}
 >
   <slot {components} />
```
Comment (DefinitelyMaybe) [inline packages/extras/src/lib/components/Instancing/InstancedMeshes/InstancedMeshes.svelte:59]: brain fart here on my part, but would this be our usual

```suggestion
  <slot ref={components} />
```
or doesn't make sense here, i dunno 😅

File: packages/extras/src/lib/components/MeshRefractionMaterial/MeshRefractionMaterial.svelte [hunk 4]
```diff
@@ -50,19 +49,19 @@
...
+  let defines: Record<string, string> = {}
 
   const updateDefines = (
-    envMap: $$Props['envMap'],
-    aberrationStrength: $$Props['aberrationStrength'],
-    fastChroma: $$Props['fastChroma']
+    envMap: CubeTexture | Texture | undefined,
+    aberrationStrength: number,
+    fastChroma: boolean
   ) => {
```
Comment (DefinitelyMaybe) [inline packages/extras/src/lib/components/MeshRefractionMaterial/MeshRefractionMaterial.svelte:61]: 🎉

**Review by grischaerbe — APPROVED** [review]
You are crazy. Incredible work 💪 

LGTM! If we catch smaller things, we'll do that in a follow-up PR 👍

**Pull request merged**